inference-engine/src/vpu/graph_transformer/src/stages/interpolate.cpp

```diff
@@ -0,0 +1,80 @@
+    IE_ASSERT(inputs.size() <= 4 && inputs.size() >= 1);
+    IE_ASSERT(outputs.size() == 1);
+
+    using ngraph::opset4::Interpolate;
+    const auto interpolateMode = _layer->GetParamAsString("mode");
+
+    const auto input = inputs[0];
+    const auto output = outputs[0];
+
+    // try to use existing resize layers
+    if (input->desc().dimsOrder() == DimsOrder::NCHW || input->desc().dimsOrder() == DimsOrder::NHWC ||
+        input->desc().dimsOrder() == DimsOrder::CHW || input->desc().dimsOrder() == DimsOrder::HWC) {
+        const auto ic = input->desc().dim(Dim::C);
+        const auto in = (input->desc().numDims() == 3) ? 1 : input->desc().dim(Dim::N);
+
+        const auto oc = output->desc().dim(Dim::C);
+        const auto on = (output->desc().numDims() == 3) ? 1 : output->desc().dim(Dim::N);
+
+        auto pads_begin = _layer->GetParamAsInts("pads_begin", {});
+        auto pads_end = _layer->GetParamAsInts("pads_end", {});
+
+        const auto isPadZeros = [](const std::vector<int>& pad) {
+            return std::all_of(pad.begin(), pad.end(), [](int i) { return i == 0; });
+        };
+
+        if (ic == oc && in == 1 && on == 1 && isPadZeros(pads_begin) && isPadZeros(pads_end)) {
+            ie::details::CaselessEq<std::string> cmp;
+            if (cmp(interpolateMode, "nearest")) {
+                // current "Resample" supports the following "Interpolate" modes only:
+                // coordinate_transformation_mode = half_pixel; nearest_mode = round_prefer_ceil;
+                // other "Interpolate" modes are translated to the default ones
+                const auto antialias = _layer->GetParamAsBool("antialias", false);
+                _stageBuilder->addResampleNearestStage(model,
+                                                       _layer->name,
+                                                       _layer,
+                                                       antialias,
+                                                       -1.0f,
+                                                       input,
+                                                       output);
+            } else if (cmp(interpolateMode, "linear")) {
+                // current "Interp" supports modes "align_corners" and "asymetric" only
+                const auto coordinate_transformation_mode = _layer->GetParamAsString("coordinate_transformation_mode", "half_pixel");
+
+                _stageBuilder->addInterpStage(model,
+                                              _layer->name,
+                                              _layer,
+                                              cmp(coordinate_transformation_mode, "align_corners"),
+                                              input,
+                                              output);
+            } else {
+                VPU_THROW_EXCEPTION << "Current Interpolate supports 'nearest' and 'linear' modes only";
```

Comment:
It would be good to mention the failed layer name for this exception and for the exceptions above.

Reply:
Fixed